src/lib/fragments/Accordion.react.js

```diff
@@ -4,42 +4,204 @@ import xtype from 'xtypejs';
+      case 'ArrowDown':
+        event.preventDefault();
+        this.handleDownArrow(index, eventOrigin);
+        break;
+      case 'ArrowUp':
+        event.preventDefault();
+        this.handleUpArrow(index, eventOrigin);
+        break;
+      default:
+        break;
+    }
+  }
+
+  findFocusableElement(element, direction) {
+    if (!element) {
+      return
+    }
+    const focusableElements = 'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex="0"], [contenteditable]';
+    const focusable = Array.from(document.querySelectorAll(focusableElements));
+    const currentIndex = focusable.indexOf(element);
+    if (currentIndex === -1) {
+      console.warn("Element not found in the list of focusable elements.");
+      return null;
+    }
+
+    let targetIndex;
+    if (direction === "next") {
+      targetIndex = currentIndex + 1;
+    } else if (direction === "previous") {
+      targetIndex = currentIndex - 1;
+    } else {
+      throw new Error("Invalid direction. Must be either 'next' or 'previous'.");
+    }
+    return focusable[targetIndex] || null;
+  }
 
   render() {
-    let accordionContent 
-    if (this.props.children.length === 1 || xtype.type(this.props.children) !== 'array'){                     
+    let accordionContent
+    if (this.props.children.length === 1 || xtype.type(this.props.children) !== 'array') {
       accordionContent = this.renderAccordionSection(0, this.props.children, this.state.sectionsOpen[0])
     }
-    else { 
+    else {
       accordionContent = this.props.children.map((accordionSectionContent, index) => this.renderAccordionSection(index, accordionSectionContent, this.state.sectionsOpen[index]))
     }
     return (
       <div className="js-enabled">
-        <div className="govuk-accordion" data-module="govuk-accordion" id={this.props.id}>
+        <div
+          className="govuk-accordion"
+          data-module="govuk-accordion"
+          id={this.props.id}
+        >
           <div className='govuk-accordion__controls'>
-            <button type='button' className='govuk-accordion__show-all' onClick={this.showOrHideAllAccordionSections}>
-            <span className= {this.state.allSectionsAreOpen ? "govuk-accordion-nav__chevron" : "govuk-accordion-nav__chevron govuk-accordion-nav__chevron--down"} ></span>
-            <span className="govuk-accordion__section-toggle-text"> {this.state.allSectionsAreOpen ? "Hide all sections" : "Show all sections"} </span>
+            <button
+              type='button'
+              className='govuk-accordion__show-all'
+              onClick={this.showOrHideAllAccordionSections}
+              onKeyDown={(event) => this.handleKeyEvent(event, -1)}
+              aria-label={`Expandable section with ${this.props.accordionHeadings.length} sections, select to ${this.state.allSectionsAreOpen ? "close" : "open"} all sections`}
```

Comment:
is "*select* to open" understandable phrasing? I don't really know if it is for screen reader users but *select* feels a little confusing to me

Reply:
I've got some ideas, I think the concise version is okay? do you have any preference? 

Concise version:
"Toggle 4 panels, expand all" (when all panels are closed)
"Toggle 4 panels, collapse all" (when all panels are open)

More verbose alternative 1:
"Click to expand or collapse 4 accordion panels, expand all" (when all panels are closed)
"Click to expand or collapse 4 accordion panels, collapse all" (when all panels are open)

More verbose alternative 2:
"Select to toggle the expansion of 4 accordion panels, expand all" (when all panels are closed)
"Select to toggle the expansion of 4 accordion panels, collapse all" (when all panels are open)

Reply:
I think the concise version works too! Thanks Andy

Reply:
Just so your aware they is no standard for this. I was likely advise again using anything (like in the official gov GDS). It's ok to say "accordion panels" but it actually provides no extra content due limited understanding of what these panels are. By default it should read "Show all sections" if im not mistaken due to the text on the button.

Reply:
PS if you ever get stuck on some aria take a look at this https://react-spectrum.adobe.com/react-aria/ Adobe done a great job of making functional react components highly accessible via hooks so you can skin and make changes to GDS standards :)

Reply:
CC: @tom-clarke-dluhc  Have a read what I put, is this still a good idea to add. I believe it will provide a negative experience